src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -137,6 +142,9 @@ def pre_fork(self):
     if self._graph_helper:
       self._graph_helper.scheduler.pre_fork()
 
+    # Acquired here, and released in both the parent and child post_fork.
+    self._fork_lock.acquire()
```

Comment:
how about just wrapping `self.daemonize` with a contextmanager form of lock acquisition:

```
def run(self):
  with self._fork_lock:
    self.daemonize(...)
```